pegasus/helpers/section_api_helpers.rb

```diff
@@ -299,16 +303,20 @@ def self.valid_scripts(user_id = nil)
   def self.valid_courses
     course_cache_key = I18n.locale.to_s
 
-    return @@course_cache[course_cache_key] if @@course_cache.key?(course_cache_key)
+    if @@course_cache.key?(course_cache_key) && !rack_env(:levelbuilder)
+      return @@course_cache[course_cache_key]
+    end
 
     return {} unless (Dashboard.db[:courses].count rescue nil)
 
-    @@course_cache[course_cache_key] = Dashboard.db[:courses].
+    courses = Dashboard.db[:courses].
       select(:id, :name).
       all.
       # Only return courses we've whitelisted in ScriptConstants
       select {|course| ScriptConstants.script_in_category?(:full_course, course[:name])}.
       map {|course| assignable_info(course)}
+    @@course_cache[course_cache_key] = courses unless rack_env(:levelbuilder)
+    courses
```

Comment:
Presumably we could write to the cache regardless (but only read from cache if !levelbuilder) if we wanted to minimize how different things are.